File: android/app/src/main/java/com/sharpreminder/triggers/model/SignalSnapshot.kt
package com.sharpreminder.triggers.model

import java.time.Instant

/**
 * État des signaux surveillés à un instant donné. Miroir Kotlin de
 * `src/domain/triggers/signal.ts`.
 *
 * Décrit le monde *maintenant*, pas ce qui vient de se produire : c'est ce qui
 * rend le ET possible entre plusieurs conditions.
 */
data class SignalSnapshot(
    val now: Instant,
    val wifiSsid: String?,
    val connectedBluetoothDeviceIds: Set<String>,
    /** Identifiants des *conditions* de lieu dont la zone est occupée. */
    val insideLocationConditionIds: Set<String>,
) {
    companion object {
        fun empty(now: Instant) = SignalSnapshot(
            now = now,
            wifiSsid = null,
            connectedBluetoothDeviceIds = emptySet(),
            insideLocationConditionIds = emptySet(),
        )
    }
}

File: ios/SharpReminder/Triggers/SignalSnapshot.swift
import Foundation

/// État des signaux surveillés à un instant donné. Miroir Swift de
/// `src/domain/triggers/signal.ts` et de `SignalSnapshot.kt`.
///
/// Décrit le monde *maintenant*, pas ce qui vient de se produire : c'est ce qui
/// rend le ET possible entre plusieurs conditions.
struct SignalSnapshot {
    let now: Date
    let wifiSsid: String?
    let connectedBluetoothDeviceIds: Set<String>
    /// Identifiants des *conditions* de lieu dont la zone est occupée.
    let insideLocationConditionIds: Set<String>

    static func empty(now: Date) -> SignalSnapshot {
        SignalSnapshot(
            now: now,
            wifiSsid: nil,
            connectedBluetoothDeviceIds: [],
            insideLocationConditionIds: []
        )
    }
}
